heudiconv/convert.py

```diff
@@ -630,6 +630,19 @@ def convert(
                         item_dicoms, prefix, with_prov, bids_options, tmpdir, dcmconfig
                     )
 
+                    # try to handle compression failures from dcm2niix
+                    if outtype == 'nii.gz':
+                        converted_files = res.outputs.converted_files
+                        if not isinstance(converted_files, list):
+                            converted_files = [converted_files]
+                        uncompressed = [x for x in converted_files if x.endswith('.nii')]
+                        if len(uncompressed) > 0:
+                            lgr.warning("Conversion returned uncompressed nifti (>4GB?) - "
+                                        "trying to salvage by recompressing ourselves. "
+                                        "This might take a while ")
+                            if not recompress_failed(uncompressed):
+                                raise RuntimeError("Error compressing nifti")
```

Comment:
why to delay this crash here?  this way you are hiding from user information on individual file and corresponding error -- why not just to just call `recompress` here and let error (if any happens) to bubble up?

I would also just loop here and have `recompress` to operate on a single file.

FWIW, I think it would be worth adding a test scenario to test this code path. I would have just mock patched the `nipype_convert` so that I would decompress some output file(s) and thus triggering this code path.

Reply:
Thanks, I'll look into it!